File: scripts/populate/scripts/objects/movie.py
from enum import Enum
from datetime import datetime,timedelta
from dataclasses import dataclass
@dataclass
class Movie: 
    class Rating(Enum): 
        PG = "PG" 
        PG_13 = "PG-13"
        #R ="R"
        G= "G"
    name : str 
    rating : Rating 
    description : str  
    duration : datetime  
    reales_date : datetime 
    poster : str 
    trailer : str  
    id : int 
    def __iter__(self): 
        """(`Name`, `rating`, `id`, `description`, `duration`, `reales_date`, `poster_images`, `trailer`)"""
        return iter((
            self.name, 
            self.rating.value , 
            self.id , 
            self.description ,
            self.duration , 
            self.reales_date.strftime("%Y-%m-%d") , 
            self.poster , 
            self.trailer
        ))
        

File: backend/cinematown/src/main/java/com/api/cinematown/Models/Movie.java
package com.api.cinematown.Models;

import jakarta.persistence.*;
import java.sql.Date;
import java.sql.Time;
import java.util.Set;

@Entity
@Table(name = "movie")
public class Movie  {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
     public Long id ;

    public String name ;
    public String rating ;

    public String description ;
    public Date reales_date ;


    private  String poster_images ;

    public String trailer ;

    public Time duration ;



    @ManyToMany
    @JoinTable(
            name = "movie_genre",
            joinColumns = @JoinColumn(name = "movie_id"),
            inverseJoinColumns = @JoinColumn(name = "genre_id"))
    Set<Genre> genres ;


    @OneToMany(mappedBy = "movie")
    Set<Schedule> schedules;



    //@JsonValue
    //public Map<String, Object> getTrailer() {
        //return JsonParserFactory.getJsonParser().parseMap(this.trailer);
    //}


    
   // @JsonValue
    //public Map<String, Object> getPoster_images() {
      //  return JsonParserFactory.getJsonParser().parseMap(this.poster_images);
    //}
}